File: AspNetWpf/SharedDTOs/Models/CustomerDto.cs
using System.ComponentModel.DataAnnotations;

namespace SharedDTOs.Models
{
    // FromBody ではコンストラクタ必須の DTO をバインドできずエラーになるため、プロパティ形式に変更
    public record CustomerDto()
    {
        [Required]
        [StringLength(10, MinimumLength = 1)]
        public string Code { get; set; } = null!;

        [Required]
        public string Name { get; set; } = null!;

        [Required]
        public string NameKana { get; set; } = null!;

        [Required]
        public string Prefecture { get; set; } = null!;
    };
}

File: AspNetAndroid/app/src/main/java/com/example/aspnetandroidsample/data/dto/CustomerDto.kt
package com.example.aspnetandroidsample.data.dto

import com.example.aspnetandroidsample.model.CustomerInfo

data class CustomerDto(
    val code: String?,
    val name: String?,
    val nameKana: String?,
    val prefecture: String?,
) {
    fun toCustomerInfo() = CustomerInfo(code ?: "", name ?: "", nameKana ?: "", prefecture ?: "")
}
